Version 7.2 - Egg system fixes and UI improvements [1.20.1 7.2

soyelchecho opened the pull request:
Major Updates:
  - Updated Forge from 47.4.0 to 47.4.12
  - Updated GeckoLib from 4.7 to 4.8.2
  - Version bump to 7.2

Egg System:
  - Fixed egg block tick system to work correctly
  - Fixed owner synchronization between server and client
  - Auto-update legacy eggs with owner information
  
UI Improvements:
  - Added hatch progress bar with color-coded stages
  - Added visual countdown timer (MM:SS format)
  - Owner name now displays correctly in egg overlay

Updates:
  - Updated to Forge 47.4.12 and GeckoLib 4.8.2
  - Fixed config initialization and deprecated API calls

soyelchecho committed:
Version 7.2 - Egg system fixes and UI improvements

Major Updates:
- Updated Forge from 47.4.0 to 47.4.12
- Updated GeckoLib from 4.7 to 4.8.2
- Version bump to 7.2

Egg System:
- Fixed egg block tick system to work correctly
- Fixed owner synchronization between server and client
- Auto-update legacy eggs with owner information

UI Improvements:
- Added hatch progress bar with color-coded stages
- Added visual countdown timer (MM:SS format)
- Owner name now displays correctly in egg overlay

Bug Fixes:
- Fixed config initialization and deprecated API calls
- Fixed client-side class loading on dedicated servers
- Fixed Jade showing "???" for owned animals

soyelchecho committed:
Add formating style with spotless and Google java format

ginerJ commented:
Hi Sergio! Thanks for the dedication put over this PR. I have read it whole and tested it a bit, will send it to the testers and get back to you. Overall a definitely enhancement of the hatching GUI and readability of the source.

soyelchecho committed:
Fix Rabbit Scared and Spawn Flying mods

soyelchecho committed:
Add optional Forge and Biomes O' Plenty biome compatibility for pterosaur spawns

Extended biome spawn tags for Argentavis, Quetzalcoatlus, and Grapteranodon to support modded biomes using optional tag entries. All Forge tags and BOP biomes use "required": false to prevent errors when mods are not installed.

Changes:
- Added optional Forge biome tags (#forge:is_mountain, #forge:is_savanna)
- Added optional c: namespace tags for future 1.21+ compatibility
- Added specific Biomes O' Plenty biomes (alps, volcano, wasteland, etc.)
- Maintains full vanilla compatibility with no breaking changes

This ensures pterosaurs spawn correctly in vanilla while also being discoverable in modded biomes when Forge biome mods or Biomes O' Plenty are installed.

ginerJ commented:
After further analysis got a crash out of trying to spawn a Ground Beetle. I suggest further testing before adding more PRs, as is something like this slipped who knows what more could get the game to crash.

I have also decided that I do like the changes you have made, but if possible I would want to avoid the reformatting changes. I believe some of the file formatting to be for the better but some make the files unreadable to my liking.

If a PR with more extensive testing and no reformatting changes with only new stuff was provided I would merge it. Nonetheless I understand if this is too tedious to ask, so I am also asking for permission to implement the changes myself if you don't wish to continue with the PRs. Thanks for your time.

Crach report:
https://pastebin.com/K0YzuLcS

soyelchecho commented:
Hey @ginerJ  Got it, let me implement it and I'll try to undo the google formating

soyelchecho commented:
@ginerJ  could you please specify the steps to reproduce the crash? I have the fix

ginerJ commented:
Thanks in advance. I interacted over the ground with the Ground Beetle Spawn Egg thus spawning the mob which instantly generated the crash.

NexelOfficial commented:
This pull request is absolutely ridiculous. Why would you change 147 files in one pull request? That is an abnormal amount of changes for the features you've described.
